File: apps/web-admin/src/api/product.ts
import apiClient from './client';
import type { ApiResponse, PageResponse } from './types';

export interface Product {
  id: number;
  name: string;
  imageUrl: string | null;
  pointCost: number;
  stock: number;
  active: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductParams {
  name: string;
  imageUrl?: string | null;
  pointCost: number;
  stock: number;
  active: boolean;
}

export type UpdateProductParams = CreateProductParams;

export const productApi = {
  getList: (page: number, size: number) =>
    apiClient.get<ApiResponse<PageResponse<Product>>>('/admin/products', {
      params: { page, size },
    }),

  getById: (id: number) =>
    apiClient.get<ApiResponse<Product>>(`/admin/products/${id}`),

  create: (data: CreateProductParams) =>
    apiClient.post<ApiResponse<Product>>('/admin/products', data),

  update: (id: number, data: UpdateProductParams) =>
    apiClient.put<ApiResponse<Product>>(`/admin/products/${id}`, data),

  delete: (id: number) =>
    apiClient.delete(`/admin/products/${id}`),
};

File: apps/backend/src/main/kotlin/com/example/pointroulette/product/entity/Product.kt
package com.example.pointroulette.product.entity

import com.example.pointroulette.common.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table

@Entity
@Table(name = "product")
class Product(
  @Column(nullable = false, length = 100)
  var name: String,

  @Column(length = 500)
  var imageUrl: String? = null,

  @Column(nullable = false)
  var pointCost: Int,

  @Column(nullable = false)
  var stock: Int,

  @Column(nullable = false)
  var active: Boolean = true,
) : BaseEntity() {

  fun update(name: String, imageUrl: String?, pointCost: Int, stock: Int, active: Boolean) {
    this.name = name
    this.imageUrl = imageUrl
    this.pointCost = pointCost
    this.stock = stock
    this.active = active
  }
}
